### PR TITLE
[MAINTENANCE] Move Markdown rendering logic to top-level notes field in Expectation and ExpectationSuite

### PR DESCRIPTION
**Pull request by cdkini**
Remove support within meta and promote to top-level notes field

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: docs/docusaurus/docs/oss/guides/expectations/advanced/how_to_add_comments_to_expectations_and_display_them_in_data_docs.md [hunk 1]
```diff
@@ -39,63 +39,33 @@ Leads to the following representation in the <TechnicalTag tag="data_docs" text=
...
-              "Second example note **without** *Markdown*",
-          ]
-      }
-  }
-)
-```
-
-![Multiple lines of string rendered without formatting](/docs/oss/images/multiple_line_string.png)
-
-
```
Comment (cdkini) [inline docs/docusaurus/docs/oss/guides/expectations/advanced/how_to_add_comments_to_expectations_and_display_them_in_data_docs.md:98]: Ignoring this feature - let's default to Markdown and if a user wants to escape md-syntax, they can

File: great_expectations/core/expectation_suite.py [hunk 2]
```diff
@@ -138,6 +139,7 @@ def __init__(  # noqa: PLR0913
         # We require meta information to be serializable, but do not convert until necessary
         ensure_json_serializable(meta)
         self.meta = meta
+        self.notes = notes
```
Comment (cdkini) [inline great_expectations/core/expectation_suite.py:142]: Suites need a notes field as well

File: great_expectations/expectations/expectation.py [hunk 3]
```diff
@@ -282,7 +282,7 @@ class Config:
 
     id: Union[str, None] = None
     meta: Union[dict, None] = None
-    notes: Union[str, None] = None
+    notes: Union[str, List[str], None] = None
```
Comment (cdkini) [inline great_expectations/expectations/expectation.py:285]: Meta notes support strings and lists of strings - we should make this consistent

File: great_expectations/expectations/expectation_configuration.py [hunk 4]
```diff
@@ -152,6 +152,7 @@ def __init__(  # noqa: PLR0913
         expectation_type: str,
         kwargs: dict,
         meta: Optional[dict] = None,
+        notes: str | list[str] | None = None,
```
Comment (cdkini) [inline great_expectations/expectations/expectation_configuration.py:155]: Don't love updating this object but we do some translation between expectation and config so we need parity

File: great_expectations/render/renderer/content_block/content_block.py [hunk 5]
```diff
@@ -318,59 +319,30 @@ def _render_expectation_meta_notes(cls, expectation):  # noqa: PLR0912
...
+                            "styling": {"parent": {"styles": {"color": "red"}}},
+                        }
+                    )
+                ]
             elif isinstance(notes, list):
-                note_content = notes
-
-            elif isinstance(notes, dict):
-                if "format" in notes:
-                    if notes["format"] == "string":
```
Comment (tyler-hoffman) [inline great_expectations/render/renderer/content_block/content_block.py:332]: love getting rid of some of this branching!

File: tests/core/test_expectation_configuration.py [hunk 6]
```diff
@@ -172,3 +172,30 @@ def test_expectation_configuration_get_evaluation_parameter_dependencies_with_qu
     # Should fully skip `nested_update` calls in method due to lacking an "expectation_suite_name" key
     dependencies = ec.get_evaluation_parameter_dependencies()
     assert dependencies == {}
+
+
+@pytest.mark.unit
+def test_expectation_configuration_to_domain_obj():
```
Comment (tyler-hoffman) [inline tests/core/test_expectation_configuration.py]: I know we hate updating the ExpectationConfiguration, but while you're here, I'd probably just parameterize this test to go through notes being string, list, none.

File: tests/render/renderer/content_block/test_content_block.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+                    "markdown": "**notes**",
+                    "styling": {"parent": {}},
+                },
+            ],
+            id="list notes",
+        ),
+    ],
+)
+@pytest.mark.unit
+def test__render_expectation_notes_with_notes(expectation, expected):
```
Comment (tyler-hoffman) [inline tests/render/renderer/content_block/test_content_block.py]: Minor, but I'd add types to the signature.

File: tests/render/test_column_section_renderer.py [hunk 8]
```diff
@@ -834,7 +844,13 @@ def test_ExpectationSuiteColumnSectionRenderer_expectation_with_single_string_me
                                     },
                                     "subheader": "Notes:",
                                     "text": [
-                                        "This is just a single string, assigned to the 'content' key of a notes dict."
+                                        {
+                                            "content_block_type": "markdown",
+                                            "markdown": "This is just a single string, assigned to the 'content' key of a notes dict.",
```
Comment (tyler-hoffman) [inline tests/render/test_column_section_renderer.py]: `assigned to the 'content' key of a notes dict` that seems to be no longer correct

File: tests/render/test_column_section_renderer.py [hunk 9]
```diff
@@ -1143,7 +1175,13 @@ def test_ExpectationSuiteColumnSectionRenderer_expectation_with_single_string_me
                                     },
                                     "subheader": "Notes:",
                                     "text": [
-                                        "This is a single string assigned to the 'notes' key."
+                                        {
+                                            "content_block_type": "markdown",
+                                            "markdown": "This is a single string assigned to the 'notes' key.",
```
Comment (tyler-hoffman) [inline tests/render/test_column_section_renderer.py]: another note that's wrong. I probably would fix it even though it's dummy text.

**Review by tyler-hoffman — CHANGES_REQUESTED** [review]
Couple minor things. Mostly I'd just update the dummy text since it talks about now-invalid data shapes.

**Review by tyler-hoffman — APPROVED** [review]

**Pull request merged**